src/dune/dune_file.ml

```diff
@@ -214,6 +214,24 @@ module Preprocess_map = struct
         List.fold_left (Preprocess.pps pp) ~init:acc ~f:(fun acc (loc, pp) ->
             Lib_name.Map.set acc pp loc))
     |> Lib_name.Map.foldi ~init:[] ~f:(fun pp loc acc -> (loc, pp) :: acc)
+
+  let add_bisect t =
+    let bisect_ppx =
+      let bisect_name = Lib_name.parse_string_exn (Loc.none, "bisect_ppx") in
+      (Loc.none, bisect_name)
+    in
+    Per_module.map t ~f:(fun pp ->
+      match pp with
+      | Preprocess.No_preprocessing ->
+        let loc = Loc.none in
+        let pps = [ bisect_ppx ] in
+        let flags = [] in
+        let staged = false in
+        Preprocess.Pps { loc; pps; flags; staged }
+      | Preprocess.Pps { loc; pps; flags; staged } ->
+        let pps = bisect_ppx :: pps in
+        Preprocess.Pps { loc; pps; flags; staged }
+      | _ -> pp (* TODO: decide if this is the correct behavior *) )
```

Comment:
Ideally, we would setup a "preprocessing" pipeline and apply both `bisect_ppx` and the custom action. But for now we can leave that as a known issue.

Reply:
Actually, since we are adding a `bisect_ppx` field on libraries and executables, we should error-out in this case.